BUILD

```diff
@@ -118,19 +118,52 @@ cc_library(
     ],
 )
 
-cc_library(
-    name = "wasmtime_lib",
+genrule(
+    name = "prefixed_wasmtime_sources",
     srcs = [
-        "src/common/types.h",
         "src/wasmtime/types.h",
         "src/wasmtime/wasmtime.cc",
     ],
+    outs = [
+        "src/wasmtime/prefixed_types.h",
+        "src/wasmtime/prefixed_wasmtime.cc",
+    ],
+    cmd =
+        """
+        for file in $(SRCS); do
+           sed -e 's/wasm_/wasmtime_wasm_/g' \
+               -e 's/wasmtime\\/types.h/wasmtime\\/prefixed_types.h/g' \
+           $$file >$(@D)/$$(dirname $$file)/prefixed_$$(basename $$file)
+        done
+        """,
+)
```

Comment:
Ideally, this shouldn't be in the top-level `BUILD` file, but I'm going to move the build rules for the runtimes/engines into packages in a future PR, and I didn't want to do the refactoring here now.